vit-servicing-station-lib/src/v0/endpoints/funds/routes.rs

```diff
@@ -23,18 +23,21 @@ pub async fn filter(
 
     let all_funds = warp::path::end()
         .and(warp::get())
-        .and(with_context.clone())
+        .and(with_context)
         .and_then(get_all_funds)
         .boxed();
 
-    let put_fund = warp::path::end()
-        .and(warp::put())
+    // fund_by_id need to be checked first otherwise requests are swallowed by the fund::any
+    root.and(fund_by_id.or(fund).or(all_funds)).boxed()
+}
+
+pub fn admin_filter(
+    context: SharedContext,
+) -> impl Filter<Extract = (impl Reply,), Error = Rejection> + Clone {
+    let with_context = warp::any().map(move || context.clone());
+
+    warp::put()
```

Comment:
nit: could use `path::end()` here? (not sure if it makes much sense though, just a though)